addons/stock/models/stock_warehouse.py

```diff
@@ -1018,6 +1018,16 @@ def _get_all_routes(self):
         routes |= self.env["stock.location.route"].search([('supplied_wh_id', 'in', self.ids)])
         return routes
 
+    def action_archive(self):
```

Comment:
Instead of override `action_archive` and `action_unarchive`, it will be safer to check in the write with the `active` field, no ? 
```python
def write():
...
   res = super(Warehouse, self).write(vals)
   if 'active' in vals:
      _check_multiwarehouse_group()
...
```